Floral Somatoray Upgrade

Aurum22 opened the pull request:
While in mutate mode, the Floral Somatoray has a 5% chance to cause a species shift (if possible).

:cl: Cronac
tweak: Floral Somatoray can now induce species shifts while in "mutate" mode.
/:cl:
<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You'll find a README and example file in .\html\changelogs\ for further instructions.

You can also find a template for adding your changelog directly to the PR description here: https://github.com/Baystation12/Baystation12/wiki/Automatic-changelog-generation
-->

Hubblenaut commented on code/modules/hydroponics/trays/tray.dm:
```diff
@@ -178,14 +178,17 @@
 			if(seed)
 				seed = seed.diverge_mutate_gene(G.gene, get_turf(loc))	//get_turf just in case it's not in a turf.
-		else
+		if(prob(75))
```
This will cause mutations when using the somatoray in gene targetting mode.

Aurum22 replied:
I made tweaks overall.

Hubblenaut commented on code/modules/hydroponics/trays/tray.dm:
```diff
@@ -178,14 +178,17 @@
 			mutate(1)
 			return
+		else
+			mutate_species(1)
```
`mutate_species()` does not take an argument.

Aurum22 replied:
Understood!

Hubblenaut commented on code/modules/hydroponics/trays/tray.dm:
```diff
@@ -178,14 +178,17 @@
 				seed = seed.diverge_mutate_gene(G.gene, get_turf(loc))	//get_turf just in case it's not in a turf.
-		else
+		if(prob(75))
 			mutate(1)
```
Just noting that `mutate()` already appears to be handling calls of `mutate_species()` if severity is greater one.
The changes could be simplified to `mutate(prob(75) ? 1 : 2`

Aurum22 replied:
Made the change and tested it; it works.

Aurum22 committed:
Floral Somatoray Tweak

Enables the floral somatoray to cause species shifts with a 5% chance while on mutate mode.

Aurum22 commented:
Made some changes based on commentary, tested the changes successfully. Did a code squish and re-submitted.
